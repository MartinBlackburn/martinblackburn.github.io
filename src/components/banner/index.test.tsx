// libraries
import { render } from "@testing-library/react";

// component under test
import Banner from ".";

describe("Banner component", () => {
    describe("snapshots", () => {
        test("with profile image", () => {
            const { asFragment } = render(
                <Banner
                    introText="// hello world"
                    titleText="I'm Martin Blackburn"
                    backgroundImage="/background.png"
                    profileImage="/profile.png"
                />
            );

            expect(asFragment()).toMatchSnapshot();
        });

        test("without profile image", () => {
            const { asFragment } = render(
                <Banner introText="// hello world" titleText="I'm Martin Blackburn" backgroundImage="/background.png" />
            );

            expect(asFragment()).toMatchSnapshot();
        });

        test("in dark mode", () => {
            const { asFragment } = render(
                <Banner
                    introText="// hello world"
                    titleText="I'm Martin Blackburn"
                    backgroundImage="/background.png"
                    darken={true}
                />
            );

            expect(asFragment()).toMatchSnapshot();
        });

        test("with child components", () => {
            const child = <div>child component</div>;

            const { asFragment } = render(
                <Banner
                    introText="// hello world"
                    titleText="I'm Martin Blackburn"
                    backgroundImage="/background.png"
                    children={child}
                />
            );

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
