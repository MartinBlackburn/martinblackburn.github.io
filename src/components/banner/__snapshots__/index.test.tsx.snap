// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component snapshots in dark mode 1`] = `
<DocumentFragment>
  <header
    class="banner banner--darken"
    style="background-image: url(/background.png);"
  >
    <div
      class="banner__text"
    >
      <div
        class="banner__intro"
      >
        // hello world
      </div>
      <h1
        class="banner__name"
      >
        I'm Martin Blackburn
      </h1>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Banner component snapshots with child components 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-image: url(/background.png);"
  >
    <div
      class="banner__text"
    >
      <div
        class="banner__intro"
      >
        // hello world
      </div>
      <h1
        class="banner__name"
      >
        I'm Martin Blackburn
      </h1>
    </div>
    <div>
      child component
    </div>
  </header>
</DocumentFragment>
`;

exports[`Banner component snapshots with profile image 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-image: url(/background.png);"
  >
    <img
      alt="profile"
      class="banner__profile"
      src="/profile.png"
    />
    <div
      class="banner__text"
    >
      <div
        class="banner__intro"
      >
        // hello world
      </div>
      <h1
        class="banner__name"
      >
        I'm Martin Blackburn
      </h1>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Banner component snapshots without profile image 1`] = `
<DocumentFragment>
  <header
    class="banner"
    style="background-image: url(/background.png);"
  >
    <div
      class="banner__text"
    >
      <div
        class="banner__intro"
      >
        // hello world
      </div>
      <h1
        class="banner__name"
      >
        I'm Martin Blackburn
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
